project_template/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.15)
+
+project(az_exampleshortname
+    LANGUAGES C CXX

```

Comment:
Should we limit it to `C` only?

Reply:
If `CXX` is removed here, then when building tests you'll need to `enable_language(CXX)`[1]. I've not actually used this command and it looks like there are some complex restrictions on exactly how it must be used -- it is probably easiest to lift the `BUILD_TESTING` option above the project directive and conditionally enable `CXX` based on that setting.

[1] https://cmake.org/cmake/help/latest/command/enable_language.html